testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/installationmanager/InstallationManagerIntegrationTestCase.java

```diff
@@ -0,0 +1,761 @@
+        String host = "primary";
+        Path target = TARGET_DIR.resolve("installation-manager.zip");
+        File source = new File(getClass().getResource("test-repo").getFile());
+        zipDir(source.toPath().toAbsolutePath(), target);
+
+        cli.sendLine("installer update --confirm --maven-repo-file=" + target + " --host=" + host);
+        String output = cli.readOutput();
+        Assert.assertTrue(output, output.contains("org.findupdates:findupdates.installed"));
+        Assert.assertTrue(output, output.contains("org.findupdates:findupdates.removed"));
+        Assert.assertTrue(output, output.contains("org.findupdates:findupdates.updated"));
+        Assert.assertTrue(output,
+                output.contains("The candidate server has been generated. To apply it, restart the server with 'shutdown --perform-installation' command."));
+
+        Assert.assertTrue(Files.exists(primaryPrepareServerDir) && Files.isDirectory(primaryPrepareServerDir));
+        Assert.assertTrue(primaryPrepareServerDir + " does not contain the expected file marker",
+                Files.list(primaryPrepareServerDir).allMatch(p -> primaryPrepareServerDir.relativize(p).toString().startsWith("server-prepare-marker-")));
+    }
+
+    @Test
+    public void updateWithDryRunMavenZipFile() throws IOException {
+        String host = "primary";
+        Path target = TARGET_DIR.resolve("installation-manager.zip");
+        File source = new File(getClass().getResource("test-repo").getFile());
+        zipDir(source.toPath().toAbsolutePath(), target);
+
+        cli.sendLine("installer update --dry-run --maven-repo-file=" + target + " --host=" + host);
+        String output = cli.readOutput();
+        Assert.assertTrue(output, output.contains("org.findupdates:findupdates.installed"));
+        Assert.assertTrue(output, output.contains("org.findupdates:findupdates.removed"));
+        Assert.assertTrue(output, output.contains("org.findupdates:findupdates.updated"));
+        Assert.assertFalse(output,
+                output.contains("The candidate server has been generated. To apply it, restart the server with 'shutdown --perform-installation' command."));
+
+        Assert.assertFalse(Files.exists(primaryPrepareServerDir));
+    }
+
+    @Test
+    public void updateUsingBlockingTimeout() throws IOException {
+        String host = "secondary";
+        assertTrue(cli.sendLine("installer update --confirm --headers={blocking-timeout=100} --host=" + host, false));
+
+        Assert.assertTrue(Files.exists(secondaryPrepareServerDir) && Files.isDirectory(secondaryPrepareServerDir));
+        Assert.assertTrue(secondaryPrepareServerDir + " does not contain the expected file marker",
+                Files.list(secondaryPrepareServerDir).allMatch(p -> secondaryPrepareServerDir.relativize(p).toString().startsWith("server-prepare-marker-")));
+    }
+
+    @Test
+    public void updateCannotBeUsedWithMavenRepoFileAndRepositories() throws IOException {
```

Comment:
Hello @yersan. I'm not sure about the reason for this restriction. Can you please elaborate it?

Reply:
Hi @marekkopecky , to ensure the artifacts in use for the current operation comes only from the supplied maven repo file. It was assumed this condition; the Maven Repo File will contain all the required artifacts to execute the current operation. We could be more flexible here.

When you are uploading a Maven repo file, the Maven repo file is unzipped and a custom repository is created to supply the artifacts coming from the unzipped file only.

Reply:
Ok, so the reason is not technical, but this restriction is rather a consequence of the use-cases. Thanks! (I don't have a permission to resolve this thread, but please consider it as resolved.

Reply:
It is a good point @marekkopecky .

And, yes, there are no technical reasons, we could be more flexible. It was assumed you will have all requirements in your Maven Zip File.

> I don't have a permission to resolve this thread, but please consider it as resolved

That's unexpected to me since you opened the question, weird. I'll resolve it.